src/Animation.elm

```diff
@@ -0,0 +1,517 @@
+are still scheduled, and is not bounded for animations that are already done.
+-}
+timeElapsed : Clock -> Animation -> TimeDelta
+timeElapsed clock ((A { start, delay_ }) as u) =
+    if isStatic u then
+        0
+
+    else
+        clock - (start + delay_) |> max 0
+
+
+{-| Get the time until the animation is done. This time may be spent animating or be part of the delay. Will be zero for animations
+that are already done.
+-}
+timeRemaining : Clock -> Animation -> TimeDelta
+timeRemaining clock ((A { start, delay_, dos, from_, to_ }) as u) =
+    if isStatic u then
+        0
+
+    else
+        let
+            duration_ =
+                dur dos from_ to_
+        in
+        start + delay_ + duration_ - clock |> max 0
+
+
+{-| Get the _current_ velocity of the animation, aproximated by looking 10ms forwards and backwards (the central
+difference). The velocity may be negative.
+-}
+getVelocity : Clock -> Animation -> Float
+getVelocity clock u =
+    let
+        backDiff =
+            animate (clock - 10) u
+
+        forwDiff =
+            animate (clock + 10) u
+    in
+    (forwDiff - backDiff) / 20
+
+
+{-| Get the start time of the animation, not accounting for delay. For animations created with `animate`, this is the
+argument that was passed. For interrupted animations, this is when the interruption occured.
+-}
+getStart : Animation -> Clock
+getStart (A a) =
+    a.start
+
+
+{-| Get the duration of the animation, not counting delay.
+-}
+getDuration : Animation -> TimeDelta
+getDuration (A { dos, from_, to_ }) =
+    dur dos from_ to_
+
+
+{-| Get the average speed of the animation.
+-}
+getSpeed : Animation -> Float
+getSpeed (A { dos, from_, to_ }) =
+    spd dos from_ to_
+
+
+{-| Get the delay of the animation.
+-}
+getDelay : Animation -> TimeDelta
+getDelay (A a) =
+    a.delay_
+
+
+{-| Get the easing function of the animation.
+-}
+getEase : Animation -> Float -> Float
+getEase (A a) =
+    a.ease_
+
+
+{-| Get the initial value of the animation.
+-}
+getFrom : Animation -> Value
+getFrom (A a) =
+    a.from_
+
+
+{-| Get the final value of the animation.
+-}
+getTo : Animation -> Value
+getTo (A a) =
+    a.to_
+
+
+{-| Equality on animations. Compared to `(==)` (which should not be used), this
+function handles the conversion of speed and duration, and start and delay. It
+also samples the easing functions, which may produce false positives (but
+usually not in practice).
+
+    equals (animation 0) (animation 0) --> True
+    equals (animation 0 |> delay 10) (animation 10) --> True
+    equals (animation 0 |> duration 1000) (animation 0 |> speed 0.001) --> True
+
+    equals (static 0) (animation 0) --> False
+    equals (animation 0 |> from -1) (animation 0) --> False
+    equals (animation 0 |> ease identity) (animation 0) --> False
+
+-}
+equals : Animation -> Animation -> Bool
+equals (A a) (A b) =
+    a.start
```

Comment:
Formatting here got broken, I think it can be fixed by wrapping each comparison with parens

Reply:
That's been bothering me for literally years! Fixed in 0792bd0